### PR TITLE
add vapor

### PR DESCRIPTION
**Pull request by tannernelson**
This is a work in progress.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,188 +15,189 @@ env:
...
-    - "TESTDIR=Scala/play2-scala"
-    - "TESTDIR=Scala/scalatra"
-    - "TESTDIR=Scala/scruffy"
-    - "TESTDIR=Scala/spray"
-    - "TESTDIR=Scala/spray-es"
-    - "TESTDIR=Scala/unfiltered"
-    - "TESTDIR=Scala/http4s"
-    - "TESTDIR=Scala/finch"
-    - "TESTDIR=Ur/urweb"
+    ### - "TESTDIR=C/libreactor"
```
Comment (martin-g) [inline .travis.yml]: I believe big part of the changes in this file should be reverted.
You are supposed just to add the new test, not to comment out all others.
I guess you did it just to make Travis build faster while testing.

Comment (tannernelson) [inline .travis.yml]: Yeah I will revert those before merging fully. I just don't wanna make travis build a million projects as I'm making tons of test commits.

Are you familiar with this project?

Comment (martin-g) [inline .travis.yml]: Oh, sorry! 
I didn't notice that the description says: "This task is work in progress". I jumped directly to see the code.

Just ask here or in the mailing list! Someone will help you quickly!

**Comment by msmith-techempower** [comment]
Could this be redone with the `.debug` folder added to a local `.gitignore`? Seems like a lot of noise in the Files Changed view and I cannot help because of it.

**Comment by tannernelson** [comment]
Unfortunately the Swift compiler is experiencing linking errors here (error 127). I assume it's one of the environment variables throwing off the compiler searching for shared libs. 

I'm including a pre-compiled version which is running successfully against the `/plaintext` route. I just need to work on adding the rest.

**Comment by tannernelson** [comment]
Possibly I could make it so only `.so` files are included which would declutter the diff.

**Comment by tannernelson** [comment]
@msmith-techempower I solved the linker error. The diff is has been greatly simplified.

**Comment by tannernelson** [comment]
@msmith-techempower Do you know why i'm getting this error?
 ```sh
The command "time ./toolset/run-ci.py verify "$TESTDIR"" exited with 1.
```

**Comment by msmith-techempower** [comment]
@tannernelson Sorry, without some more context, I cannot provide much help. Could you link a [gist](https://gist.github.com/) of the entire log?

**Comment by tannernelson** [comment]
@msmith-techempower are you able to view the CI logs from my travis build? It's at the very end of that log. It looks like the benchmarks are running well, they even print the times, but then that error comes up at the very end.

**Comment by cjnething** [comment]
There seems to be quite a few errors before that last line that you mentioned. Fortune, db, update, and query are all failing ("WARN"s are okay, but "FAIL"s are not). Also, I don't see Swift being installed anywhere.

File: frameworks/Swift/vapor/setup.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+# install swiftenv for managing Swift versions
+eval "$(curl -sL https://gist.githubusercontent.com/kylef/5c0475ff02b7c7671d2a/raw/02090c7ede5a637b76e6df1710e83cd0bbe7dcdf/swiftenv-install.sh)"
```
Comment (tannernelson) [inline frameworks/Swift/vapor/setup.sh]: a

Comment (tannernelson) [inline frameworks/Swift/vapor/setup.sh]: @cjnething This installs `swiftenv` which will then automatically install Swift when `swift build` is ran.

**Comment by tannernelson** [comment]
@cjnething Ah, okay. That makes sense. I didn't realize failing the tests failed the build. I thought those would just register as `N/A`. 

I will get to work on supporting those tests! :)

Do you have any idea when the next benchmark will take place so I can try to get Vapor ready before then?

**Comment by cjnething** [comment]
No worries, there's a lot going on in this project! The Round 13 timeline is laid out [here](https://groups.google.com/forum/#!topic/framework-benchmarks/AGHUyMnR_0o), so if we stay on schedule then the last day to submit would be on Friday, April 15th. If you want to see Round 12's data, that's [here](https://www.techempower.com/benchmarks/). The preview data isn't up yet, but when it is it will be [here](http://tfb-logs.techempower.com/). Feel free to ask any other questions, that's what we're here for!

File: frameworks/Swift/vapor/setup.sh (hunk 2 above)
Comment (cjnething) [inline frameworks/Swift/vapor/setup.sh]: I pulled down  your repo and tested it, and that line doesn't install swift (or it's not installing correctly, at least). When it gets down to line 10 ("swift build -c release"), it says "swift: command not found"

Comment (tannernelson) [inline frameworks/Swift/vapor/setup.sh]: That's strange. The script is specifically for CIs like Travis so maybe it won't work on your local environment. The other option is installing `swiftenv` manually: https://github.com/kylef/swiftenv

Comment (cjnething) [inline frameworks/Swift/vapor/setup.sh]: Oh that's weird, I just checked the Travis build and it seems to be installing fine there. Not sure what's up with my local version then.

**Comment by tannernelson** [comment]
@cjnething Awesome! Hopefully we can make the cutoff there. Can you point me to where I should look in the Travis build logs for what scores Vapor is getting currently? I see a few scattered time stamps but I'm not 100% sure what they mean.

**Comment by cjnething** [comment]
Oh I see what you're saying. I don't think Travis gives that kind of information. When we do the official rounds, they're tested on actual hardware (as detailed [here](https://www.techempower.com/benchmarks/#section=environment&hw=peak&test=json)). Travis is just an added precaution to make sure everything is working, but the official rounds don't use Travis. On a similar note, if your Swift installation process is for CIs, I'm not sure it will work for the official round. Your best bet is to make sure it works locally and on Travis, and then it will *most likely* work for the official round (assuming there aren't any strange updates, etc.). Let me know if that doesn't make sense, it's a bit confusing to type out.

**Comment by tannernelson** [comment]
Okay that makes sense. I'll update it to install Swift manually. Thanks for your help! :)

**Comment by cjnething** [comment]
No problem!

**Comment by cjnething** [comment]
@tannernelson Hi Tanner, how are the Vapor tests coming along? Do you have any questions/anything I can help you with?

**Comment by tannernelson** [comment]
Going to work on this soon. Do you know when the deadline for the next round is?

**Comment by cjnething** [comment]
Ok great! Not sure yet. We're taking bugfixes (not new frameworks) for Round 13 and new frameworks for Round 14 right now, but we don't have a set deadline yet for Round 14. I'll let you know if that changes anytime soon. As always, if you have any questions feel free to reach out! Good luck!

**Comment by nbrady-techempower** [comment]
@tannernelson it looks like we're close to doing preview runs for round 13 and then we will start testing against round 14. I would love to see this get added to the framework suite! Please let us know if there's anything else we can help you with in the meantime!

**Comment by tannernelson** [comment]
Do the database queries need to be MySQL? We only have SQLite and MongoDB right now.

**Comment by cjnething** [comment]
We do support MongoDB, but not SQLite. You can see the setup scripts for all of the databases we support [here](https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/toolset/setup/linux/database.sh). If you're interested in seeing an example of a framework implementing MongoDB in TFB, [undertow](https://github.com/TechEmpower/FrameworkBenchmarks/tree/master/frameworks/Java/undertow) is a good example, or you can see all of the frameworks on TFB that use MongoDB [here](https://www.techempower.com/benchmarks/#section=data-r12&hw=peak&test=db&d=1). Let us know if there is anything else we can help you with!

**Comment by nbrady-techempower** [comment]
@tannernelson also as an fyi we're restructuring branches a bit so that we can start doing testing/work on future rounds in a more streamlined way. If you have some time, would you mind opening a new PR for this to target the round-14 branch?

Here's more information: http://frameworkbenchmarks.readthedocs.io/en/latest/Development/Contributing-Guide/#github-pull-request-procedure

**Comment by tannernelson** [comment]
Sure thing. We just got MySQL support working so I think I'll use that for the test.

When is the next round?

**Comment by nbrady-techempower** [comment]
@tannernelson Not exactly sure on a time frame for round-14. It seems like we're trying to get a lot in, including windows toolset support. Either way, you're welcome to leave this as a work in progress and target round-14 or 15.

**Comment by nbrady-techempower** [comment]
hey @tannernelson! Hope all is well. I see that you're working on doing some self benchmarking and haven't had time here. It'd be awesome to see this in the mix at some point in the future! I'm going to close this for now. If you do go back to this, feel free to re-open. You can also turn travis on in your forked copy to get quick feedback and can always ping any of us in the future! Good luck!

**Pull request closed without merging**